### DISCUSSION TITLE
递归改写成迭代的通用方法 - XX’s Blog

### DISCUSSION THREAD

**Discussion by giscus** [xxchan/xxchan.github.io#29] [Announcements]
# 递归改写成迭代的通用方法 - XX’s Blog

递归是像呼吸一般自然的事情。    ——罗宸《谈递归（一）：递归的五种定式》

https://xxchan.github.io/cs/2022/02/15/recursion-to-iteration.html

**Comment by beyondblog** [comment]
这个氧气面罩不错😄️

**Comment by lyfer233** [comment]
```cpp
def inorder_traversal(root):
    stack = []
    stack.append(StackFrame(param, 0))
```
结合上下文，第三行应该是`stack.append(StackFrame(root, 0))`吧

**Comment by Endle** [comment]
TLDR 这篇文章介绍的内容是用python模拟一个调用栈。请问我的理解对吗？

**Reply by xxchan** [reply]
（这篇文章已经很短了没必要 TLDR 吧）
更准确地说是：模拟一个调用栈，**目的**是改写递归。
Python 也没什么特别的，和伪代码差不多。

**Comment by taimoon** [comment]
Continuation Passing Style is all you need (probably)
